index.bs

```diff
@@ -155,21 +155,28 @@ Note, the above algorithm:
 --------
 
-Note: This pragma *appends* [=client hints token=]s to the [=environment settings object=]'s [=environment settings object/client hints set=]. It *does not* add those hints to the [=Accept-CH cache=].
-
-1. If the <{meta}> element is not a child of a <{head}> element, then return.
-2. If the <{meta}> element has no <{meta/content}> attribute, or if that attribute's value is the empty string, then return.
-3. Let |settingsObject| be the <{meta}> element's [=relevant settings object=].
-4. If the result of executing [$Is an environment settings object contextually secure?$] on |settingsObject| is `"Not Secure"`, then return.
-5. Let |browsingContext| be |settingsObject|'s  [=environment settings object/global object=]'s [=Window/browsing context=].
-6. If the [=top-level browsing context=] does not equal |browsingContext|, abort these steps.
-7. Let |acceptCHValue| be the <{meta}> element's <{meta/content}> attribute's value.
-8. Parse |acceptCHValue| according to the [=Accept-CH=] header parsing rules, as a [=field-name=].
-     [=set/Append=] each parsed [=client hints token=] to |settingsObject|'s [=environment settings object/client hints set=].
-
+Note: This metadata *appends* [=client hints token=]s to the [=environment settings object=]'s [=environment settings object/client hints set=]. It *does not* add those hints to the [=Accept-CH cache=]. After this algorithm runs once, no further modification to the [=environment settings object/client hints set=] can occur without reloading the page.
+
+1. Let |metaElement| be the <{meta}> element.
+2. If |metaElement| is not a child of a <{head}> element, then return.
+3. If |metaElement| has no <{meta/content}> attribute, or if that attribute's value is the empty string, then return.
+4. If |metaElement| has any <a href="https://dom.spec.whatwg.org/#concept-tree-preceding">preceding</a> <a href="https://dom.spec.whatwg.org/#concept-tree-sibling">sibling</a> <{link}> or <{script}> elements (or if any <{link}> or <{script}> elements have begun to execute), then return.
+5. Let |settingsObject| be |metaElement|'s [=relevant settings object=].
+6. Let |permissionsPolicy| be |metaElement|'s node document's [=Document/permissions policy=]
+7. If the result of executing [$Is an environment settings object contextually secure?$] on |settingsObject| is `"Not Secure"`, then return.
+8. Let |browsingContext| be |settingsObject|'s  [=environment settings object/global object=]'s [=Window/browsing context=].
+9. If the [=top-level browsing context=] does not equal |browsingContext|, abort these steps.
+10. Let |policyDirective| be the result of running [$ parse-policy-directive $] on |acceptCHValue| and the [=origin=] of the |metaElement|'s node document.
+11. For each |feature|->|allowlist| of |policyDirective|:
+  1. If |feature| is not a [=client hints token=], then continue.
+  2. For each |allow| of |allowlist|:
+    1. If |allow| is the special value `*`, then continue.
```

Comment:
I'm pretty sure that if any `*` was seen during parsing, then all other items would have been consumed by it, and |allowlist| would only contain the one item. That is, "`'self' * https://origin.example`" is treated exactly as "`*`"

Reply:
Ah, good point